### DISCUSSION TITLE
Code Analysis and Generation

### DISCUSSION THREAD

**Discussion by fptmark** [imartinez/privateGPT#349] [Ideas]
I'm interested in creating a model that will analyze and write code similar to what ChatGPT does today?  How can I get it to understand new file extensions (*.java, *.js, etc). From there,  I'd like to create a base training model using OSS that I can point it to on the web (e.g. - github) and then extended the model using code I've personally written.  Any guidance would be appreciated.

**Comment by drwbns** [comment]
I was thinking about the same thing for c++ code. My first question is: Does privateGPT allow for understanding references from one file to another? That would be a very necessary understanding for any type of language that uses multiple files. I haven't yet installed private GPT or looked at any of it's code but I would assume that there is a file that controls the supported extensions. Might be as simple as adding your necessary types to at least recognize some coding file types, unless that is specifically disabled so that private GPT doesn't read itself or something like that.
